app/models/resource_action.rb

```diff
@@ -71,24 +80,35 @@ def ae_uri
     uri
   end
 
-  def deliver_to_automate_from_dialog(dialog_hash_values, target, user, task_id = nil)
+  def deliver_queue(dialog_hash_values, target, user, task_id = nil)
     _log.info("Queuing <#{self.class.name}:#{id}> for <#{resource_type}:#{resource_id}>")
-    MiqAeEngine.deliver_queue(automate_queue_hash(target, dialog_hash_values[:dialog], user, task_id),
-                              :zone     => target.try(:my_zone),
-                              :priority => MiqQueue::HIGH_PRIORITY,
-                              :task_id  => "#{self.class.name.underscore}_#{id}")
+    if workflow
+      workflow.execute(:userid => user.userid, :inputs => dialog_hash_values)
+    else
```

Comment:
NOTE I don't like the `if workflow / else` I'd like to use STI and subclasses, there are a number of places which assume that dialog_field.resource_action is auto-initialized before setting any properties which screws with the subclassing and will take some deeper refactoring.